server/api/region.go

```diff
@@ -14,23 +14,57 @@
 package api
 
 import (
+	"fmt"
 	"net/http"
 	"strconv"
+	"strings"
 
 	"github.com/gorilla/mux"
 	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/pd/server"
+	"github.com/pingcap/pd/server/core"
 	"github.com/unrolled/render"
 )
 
 type regionInfo struct {
-	Region *metapb.Region `json:"region"`
-	Leader *metapb.Peer   `json:"leader"`
+	ID          uint64              `json:"id"`
+	StartKey    string              `json:"start_key"`
+	EndKey      string              `json:"end_key"`
+	RegionEpoch *metapb.RegionEpoch `json:"epoch,omitempty"`
+	Peers       []*metapb.Peer      `json:"peers,omitempty"`
+
+	Leader          *metapb.Peer      `json:"leader,omitempty"`
+	DownPeers       []*pdpb.PeerStats `json:"down_peers,omitempty"`
+	PendingPeers    []*metapb.Peer    `json:"pending_peers,omitempty"`
+	WrittenBytes    uint64            `json:"written_bytes,omitempty"`
+	ReadBytes       uint64            `json:"read_bytes,omitempty"`
+	ApproximateSize int64             `json:"approximate_size,omitempty"`
+}
+
+func newRegionInfo(r *core.RegionInfo) *regionInfo {
+	if r == nil {
+		return nil
+	}
+	return &regionInfo{
+		ID:          r.Id,
+		StartKey:    strings.Trim(fmt.Sprintf("%q", r.StartKey), "\""),
+		EndKey:      strings.Trim(fmt.Sprintf("%q", r.EndKey), "\""),
```

Comment:
what is the output if start or end key is empty?

Reply:
empty string.

Reply:
like this `"end_key": ""`.